modules/core/src/main/scala-3/scalacache/memoization/Macros.scala

```diff
@@ -0,0 +1,82 @@
+package scalacache.memoization
+
+import scala.concurrent.duration.Duration
+import scala.quoted.*
+
+import scalacache.{Cache, Flags}
+
+object Macros {
+  def memoizeImpl[F[_], V](
+      ttl: Expr[Option[Duration]],
+      f: Expr[V],
+      cache: Expr[Cache[F, V]],
+      flags: Expr[Flags]
+  )(using Quotes, Type[F], Type[V]): Expr[F[V]] =
+    commonMacroImpl(cache, keyName => '{ ${ cache }.cachingForMemoize(${ keyName })(${ ttl })(${ f })(${ flags }) })
+
+  def memoizeFImpl[F[_], V](
+      ttl: Expr[Option[Duration]],
+      f: Expr[F[V]],
+      cache: Expr[Cache[F, V]],
+      flags: Expr[Flags]
+  )(using Quotes, Type[F], Type[V]): Expr[F[V]] =
+    commonMacroImpl(cache, keyName => '{ ${ cache }.cachingForMemoizeF(${ keyName })(${ ttl })(${ f })(${ flags }) })
+
+  private def commonMacroImpl[F[_], V](
+      cache: Expr[Cache[F, V]],
+      keyNameToCachingCall: Expr[String] => Expr[F[V]]
+  )(using Quotes, Type[F], Type[V]): Expr[F[V]] = {
+    import quotes.reflect.*
+    val sym: Symbol = Symbol.spliceOwner
+
+    def hasCacheKeyExcludeAnnotation(s: Symbol): Boolean = s.annotations.exists {
+//      case Apply(Select(New(Ident("cacheKeyExclude")), _), _) => true // Don't know why this doesn't match...
```

Comment:
funny, `summon[TypeTest[Tree, Ident]].unapply(o)` gives me a None too. Guess that's what's used, since `Ident` is a type member, thus prone to type erasure, so `TypeTest` will be used underneath (like `ClassTag` would be used in scala 2).

No idea why it doesn't match though. Maybe it's worth reporting somewhere...